component/prometheus/globalrefmap.go

```diff
@@ -79,6 +79,10 @@ func (g *GlobalRefMap) GetOrAddGlobalRefID(l labels.Labels) uint64 {
 	g.mut.Lock()
 	defer g.mut.Unlock()
 
+	if l == nil {
+		return 0
+	}
```

Comment:
Can we add a comment here explaining why we return 0 here? Presumably it's because a nil label set indicates a dropped series, right?

Reply:
Moreso a guard against bad input and not pollute the cache. The bad input could be a dropped series.